core/src/main/java/org/apache/iceberg/rest/ResourcePaths.java

```diff
@@ -61,6 +61,11 @@ public String tables(Namespace ns) {
     return SLASH.join("v1", prefix, "namespaces", RESTUtil.encodeNamespace(ns), "tables");
   }
 
+  public String register(Namespace ns) {
+    return SLASH.join(
+        "v1", prefix, "namespaces", RESTUtil.encodeNamespace(ns), "tables", "register");
```

Comment:
we've had some internal discussion on what the pathing should look like and `/v1/{prefix}/namespaces/{namespace}/register` makes probably the most sense here. 
Having `/v1/{prefix}/namespaces/{namespace}/tables/register` would conflict when a table is called `register`